src/vs/workbench/contrib/mcp/common/mcpLanguageModelToolContribution.ts

```diff
@@ -222,14 +224,34 @@ class McpToolImplementation implements IToolImpl {
 				}
 			}
 
-			// Rewrite image rsources to images so they are inlined nicely
-			const addAsInlineData = (mimeType: string, value: string, uri?: URI) => {
+			// Rewrite image resources to images so they are inlined nicely
+			const addAsInlineData = async (mimeType: string, value: string, uri?: URI): Promise<VSBuffer | void> => {
 				details.output.push({ type: 'embed', mimeType, value, uri });
 				if (isForModel) {
-					result.content.push({
-						kind: 'data',
-						value: { mimeType, data: decodeBase64(value) }
-					});
+					let finalData: VSBuffer;
```

Comment:
The `value` here is always going to be a base64 string, not a URI, so you can probably pass it directly into `this._imageResizeService.resizeImage(decodeBase64(value).buffer, mimeType)`